content/updates/terralith-monolithic-terraform-architecture.md

```diff
@@ -0,0 +1,107 @@
+1. [Complexities with the 3 M’s: Multi-Environment, Multi-Region, Multi-Account](#complexities-with-the-3-ms-multi-environment-multi-region-multi-account)
+2. [Collaborating in a Terralith](#collaborating-in-a-terralith)
+3. [State File Bloat -> Plans + Applies Slow Down](#state-file-bloat-plans-and-applies-slow-down)
+4. [Blast Radius: Walking Through a Minefield](#blast-radius-walking-through-a-minefield)
+
+###### Complexities with the 3 M’s: Multi-Environment, Multi-Region, Multi-Account
+One of the primary challenges is environment isolation. With all infrastructure configuration in one place, separating resources for different environments is difficult. This  leads to a higher risk of unintended cross-environmental impacts where changes meant for one environment inadvertently affect others.
+
+In any non-trivial infrastructure, there are more variables than just the environment (production, staging, development). There are also [multiple regions](https://aws.amazon.com/about-aws/global-infrastructure/regions_az/) (e.g. AWS’ US East, US West, etc.) and multiple accounts (artifacts, log archive, disaster recovery, etc). With all these intertwined, the Terralith IaC pattern becomes prone to errors and misconfigurations. It becomes difficult to understand the relationships and dependencies between resources.
+
+In the context of fitting a city into a single skyscraper analogy, this is like trying to fit residential areas, industrial zones, commercial districts, and corporate headquarters all into different floors of the same building. It certainly is possible, but it becomes a nightmare to manage - think about all the noise complaints from the industrial zones! And it certainly is not easy to scale when this city’s population expands.
+
+###### Collaborating in a Terralith
+Collaboration in a Terralith setup can be challenging as well. Since all the resources are provisioned with one root module, state is stored in one file. A common best practice in TF is to use [state locking](https://developer.hashicorp.com/terraform/language/state/locking), which locks the state file so only one operation can be executed at a given time. This is intended to prevent odd drift scenarios and state file corruption.
+
+Because of this practice, two engineers working on completely different infrastructure in the Terralith at the same time can find themselves unable to perform state operations concurrently.
+
+Here is a diagram of what that might look like in practice. This highlights the hit to engineering productivity that the Terralith causes:
+
+![Collaboration in a Terralith](/img/updates/terralith/terralith-collaboration.png) <!-- Made with Excalidraw -->
+
+###### State File Bloat: Plans and Applies Slow Down
+Imagine a Terralith’s state file like a single, massive spreadsheet tracking every item in a rapidly growing warehouse. Eventually, it becomes so large that opening it or updating it takes forever.
```

Comment:
```suggestion
Imagine a Terralith’s state file like a single, massive spreadsheet tracking every item in a rapidly growing warehouse. Eventually, it becomes so large that comparing the current state of the resources with the state recorded in the file or updating it takes forever.
```
We rarely open state files; I'd rather focus on slowing down the refresh operation.